flake.nix

```diff
@@ -0,0 +1,210 @@
+# you found our nix flake!
```

Comment:
Maybe put most of the PR description in this file as a big comment.
Also why it's called flake.nix instead of semgrep.nix?

Reply:
added it to the file. flake.nix is the name that most nix related things expect. Since a flake can export multiple programs and libraries (i.e. semgrep, osemgrep etc.) it's also better to call it one thing